test/unit/meta/src/CudaVectorArrayWrapperTest.cpp

```diff
@@ -21,6 +22,43 @@
 
 #    include <type_traits>
 
+/* The tests here use equals for comparing float values for exact equality. This is not
+ * an issue of arithmetics. We are testing whether the values saved in a container are the same as the ones retrieved
+ * from it afterwards. In this case, returning a value that would not be exactly but only approximately equal to the
+ * one that was stored in the container would be a grave error.
+ */
+template<typename T1, typename T2>
+ALPAKA_FN_INLINE ALPAKA_FN_HOST_ACC bool equals(T1 a, T2 b)
+{
+    return a == static_cast<T1>(b);
+}
+
+template<>
+ALPAKA_FN_INLINE ALPAKA_FN_HOST_ACC bool equals<float, float>(float a, float b)
+{
+    return alpaka::math::floatEqualExactNoWarning(a, b);
+}
+
+template<>
+ALPAKA_FN_INLINE ALPAKA_FN_HOST_ACC bool equals<double, double>(double a, double b)
+{
+    return alpaka::math::floatEqualExactNoWarning(a, b);
+}
+
+// floatEqualExactNoWarning needs both arguments to be the same type. Since we want to test many types, including
+// integers, in the same fashion, we can use the integer literal on the right hand side - that will be accepted with
+// any other variable on the left hand side for the generic `==` implementation above. For the floats, we do a cast.
+template<>
+ALPAKA_FN_INLINE ALPAKA_FN_HOST_ACC bool equals<float, int>(float a, int b)
+{
+    return alpaka::math::floatEqualExactNoWarning(a, static_cast<float>(b));
+}
+
+template<>
+ALPAKA_FN_INLINE ALPAKA_FN_HOST_ACC bool equals<double, int>(double a, int b)
+{
+    return alpaka::math::floatEqualExactNoWarning(a, static_cast<double>(b));
+}
```

Comment:
Isn't that a bit overengineered? You could just compare against FP literals in the code below or just cast to the FP type.

Reply:
Right again. I forgot about T{0} which renders this contraption useless